docs/source/DeveloperGuide/OAuth2Authorization/index.rst

```diff
@@ -0,0 +1,194 @@
+----------
+
+ - :py:class:`~DIRAC.FrameworkSystem.private.authorization.grants` contains helper classes with descriptions of the flows to get and revoke an access token.
+ - :py:class:`~DIRAC.FrameworkSystem.private.authorization.utils` contains helper classes with main OAuth2 object descriptions and helper methods.
+ - :py:class:`~DIRAC.FrameworkSystem.private.authorization.AuthServer` inherit `authlib.oauth2.AuthorizationServer` and simulates the operation of OAuth 2 authorization server.
+
+
+Configuration
+-------------
+
+*Authorization Server metadata*:
+
+  DIRAC AS should contain a `metadata <https://datatracker.ietf.org/doc/html/rfc8414>`_ that an OAuth client can use to obtain the information needed to interact with DIRAC AS, including its endpoint locations and authorization server capabilities.
+  But you don't have to worry about that, just define the `/DIRAC/Security/Authorization/issuer` option in the DIRAC configuration, and everything else will be determined for you by the :py:meth:`~DIRAC.FrameworkSystem.private.authorization.utils.Utilities.collectMetadata` method.
+
+*Authorization clients*:
+
+  OAuth defines two types of `clients <https://tools.ietf.org/html/rfc6749#section-2.1>`_:
+
+   - confidential clients
+   - public clients
+
+  DIRAC AS takes both into account and already has a default *public client* (see :py:class:`~DIRAC.FrameworkSystem.private.authorization.utils.Clients`) configured to authorize DIRAC client installations via the device code authorization flow mentioned earlier.
+  The new `authorization client metadata <https://datatracker.ietf.org/doc/html/rfc7591#section-2>`_ can be described in the `/DIRAC/Security/Authorization/Clients` section in format::
+
+      CLIENT_NAME
+      {
+        client_id=MY_CLIENT_ID
+        client_secret=MY_CLIENT_SECRET
+        scope=supported scopes separated by a space
+        response_types=device,
+        grant_types=refresh_token,
+      }
+
+*Supported scopes*:
+
+  For DIRAC-specific authorization, support for the following scopes is implemented:
+
+    - `g:<DIRAC group name>` this parametric scope allows you to notify which group the user selects when logging in.
+    - `proxy` scope informs that the user expects to receive a proxy certificate instead of a token after successful authorization.
+    - `lifetime:<proxy life time in a seconds>` scope informs how long the proxy should be.
+
+
+Commands
+========
+
+Two commands were created for interaction with DIRAC AS:
+
+ - :ref:`dirac-login`
+ - :ref:`dirac-logout`
+
+Also added the ability to authorize without a certificate while configuring the DIRAC client with the :ref:`dirac-configure` command and a special ``--login`` flag.
+
+
+Authorization API
+=================
+
+With a new system component - :ref: `APIs <apis>`, was created Authorization API for =Framework= system (see :py:class:`~DIRAC.FrameworkSystem.API.AuthHandler`) which provides the necessary endpoints for interaction with DIRAC AS.
```

Comment:
`=Framework=`? I think you meant `*Framework*`. (md vs rst :) )

Reply:
yes, thanks. Ill fix it